pkg/ratelimiter/reconciler.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratelimiter
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/ratelimiter"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// A Reconciler rate limits an inner, wrapped Reconciler. Requests that are rate
+// limited immediately return RequeueAfter: d without calling the wrapped
+// Reconciler, where d is imposed by the rate limiter.
+type Reconciler struct {
+	name  string
+	inner reconcile.Reconciler
+	limit ratelimiter.RateLimiter
+}
```

Comment:
I could see a pretty good case for this being in `pkg/reconciler` package instead. We have a practice of using subpackages there to avoid collisions so it could either be `ratelimited.Reconciler` in `pkg/reconciler/ratelimited` or we could make `pkg/reconciler` a package and have it be `reconciler.RateLimited`. No strong feelings really - here feels pretty good too.